src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -108,10 +122,29 @@ int kds_add_command(struct kds_command *xcmd)
 	return err;
 }
 
-int kds_init_client(struct kds_client *client)
+int kds_init_client(struct kds_sched *kds, struct kds_client *client)
 {
+	struct kds_ctx_info info;
+	struct kds_ctrl *ctrl;
+
+	client->pid = get_pid(task_pid(current));
+	client->ctrl = kds->ctrl;
+	mutex_init(&client->lock);
+
+	/* Initial controller context private data */
+	ctrl = client->ctrl[KDS_CU];
+	info.flags = CU_CTX_OP_INIT;
+	if (ctrl)
+		ctrl->control_ctx(ctrl, client, &info);
+
 	init_waitqueue_head(&client->waitq);
 	atomic_set(&client->event, 0);
+
```

Comment:
seems we don't need atomic_set but just (client->event = 0) should be good, right?

Reply:
Yes. Will fix.

Reply:
We can't directly assign 0 to atomic_t.

Reply:
As discussed offline, we observed polling cause performance issue. Let's review this part later.